packages/commons/src/typeUtils.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * Returns true if the passed value is a record (object).
+ *
+ * @param value The value to check
+ */
+const isRecord = (value: unknown): value is Record<string, unknown> => {
+  return (
+    Object.prototype.toString.call(value) === '[object Object]' &&
+    !Object.is(value, null)
+  );
+};
+
+/**
+ * Check if a value is a string.
+ *
+ * @param value The value to check
+ */
+const isString = (value: unknown): value is string => {
+  return typeof value === 'string';
+};
+
+/**
+ * Check if a value is a number.
+ *
+ * @param value The value to check
+ */
+const isNumber = (value: unknown): value is number => {
+  return typeof value === 'number';
+};
+
+/**
+ * Check if a value is an integer number.
+ *
+ * @param value The value to check
+ */
+const isIntegerNumber = (value: unknown): value is number => {
+  return isNumber(value) && Number.isInteger(value);
+};
+
+/**
+ * Check if a value is truthy.
+ *
+ * @param value The value to check
+ */
+const isTruthy = (value: unknown): boolean => {
```

Comment:
`NaN` and functions would not pass here, but I am not sure we need them, just asking if left out this edge cases on purpose.

Reply:
True.

I don't think we need to include them as they're not JSON serializable anyway.

If in the future we need to check truthiness of these values we can still expand this function.

Reply:
> they're not JSON serializable anyway. 

makes sense